### PR TITLE
client api to link to a pre-filled workflow

### PR DESCRIPTION
**Pull request by dagda1**
## Hey, I just made a Pull Request!

Following up on the discussion on #13735 where we discussed how to provide a way of linking to a workflow with pre-filled values, I have created both a component to render the link `PrefilledWorkflowLink` and a hook that creates the `href` called `usePrefilledWorkflowHref`.

I added the hook so you can get the `href` to use with `navigate` or some other programmatic way of linking to a workflow.

A lot of the plumbing is already in place for this task, so this is hopefully pretty unobtrusive.

I would appreciate some feedback on this and if it looks good, I would like to know where I can add some docs.

#### :heavy_check_mark: Checklist

- [X] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [X] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [X] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by taras** [comment]
@Rugvip, there is more in this PR than we originally discussed. It doesn't export the Ref. After implementing this, we realize it's quite straightforward. Do you want us to change the PR to only export the ref?

**Comment by taras** [comment]
Let's keep it warm :)

**Comment by dagda1** [comment]
@benjdlambert would it be possible to get some feedack on this PR please?

File: plugins/scaffolder/src/components/PrefilledWorkflowLink/usePrefilledWorkflowHref.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+ * Hook that creates prefilled workflow url that can be used programmatically in something like navigate
+ *
+ * @public
+ */
+export function usePrefilledWorkflowHref({
+  templateName,
+  namespace,
+  formData,
+}: PrefilledWorkflowProps): string {
+  const templateRoute = useRouteRef(selectedTemplateRouteRef);
```
Comment (Rugvip) [inline plugins/scaffolder/src/components/PrefilledWorkflowLink/usePrefilledWorkflowHref.ts:31]: We can't export the route ref here, not even indirectly like this. RouteRefs are internal and need to be linked together with other sources through ExternalRouteRefs. There's one exception to this, which is the entity page route ref over [here](https://github.com/backstage/backstage/blob/dd6fa27a9f1d0f6335585c2a00e83cc3e21fcee6/plugins/catalog-react/src/routes.ts#L31), but as you can see there's some extra stuff needed to make that work, and it needs to be a very stable route ref.

I'd say that in the end the current routing system doesn't support what you're aiming for here, and I think we need to step back and figure out how one would do this in a clean way, extending the framework as necessary. I'm thinking that for now we might settle for documenting this pattern? It's not a lot of code to be repeating.

**Review by Rugvip — CHANGES_REQUESTED** [review]
Thank you! 👍 

Sorry this is a bit of a tricky one and it's been a busy couple of weeks

File: plugins/scaffolder/src/components/PrefilledWorkflowLink/usePrefilledWorkflowHref.ts (hunk 1 above)
Comment (dagda1) [reply plugins/scaffolder/src/components/PrefilledWorkflowLink/usePrefilledWorkflowHref.ts:31]: @Rugvip thank you for the reply.  What is a `stable route ref` in this context?

Would the goal still be to try and get a reference to the selected template, something like:

```ts
const templateRouteRef = getOrCreateGlobalSingleton(
  'catalog:template-route-ref',
  () =>
    createSubRouteRef({
      id: 'scaffolder/selected-template',
      parent: ??,
      path: '/templates/:namespace/:templateName',
    }),
);
```

**Comment by dagda1** [comment]
I'm going to close this for now as it requires more thought.

**Pull request closed without merging**